File: apps/tizen-tv/src/ui/watchEntryFormat.ts
import type { WatchEntry } from "../data/watchHistory.js";
import { isInProgress } from "../data/watchHistory.js";

export function formatWatchTime(ms: number): string {
  if (ms <= 0) return "0:00";
  const totalSeconds = Math.floor(ms / 1000);
  const hours = Math.floor(totalSeconds / 3600);
  const minutes = Math.floor((totalSeconds % 3600) / 60);
  const seconds = totalSeconds % 60;
  const mm = String(minutes).padStart(2, "0");
  const ss = String(seconds).padStart(2, "0");
  return hours > 0 ? `${hours}:${mm}:${ss}` : `${minutes}:${ss}`;
}

export function continueLabel(entry: WatchEntry): string {
  if (entry.type === "MOVIE") {
    return `Συνέχισε · ${formatWatchTime(entry.positionMs ?? 0)}`;
  }
  if (entry.type === "SERIES_EPISODE") {
    const base = entry.subtitle?.trim() || "Επεισόδιο";
    return `${base} · ${formatWatchTime(entry.positionMs ?? 0)}`;
  }
  return entry.title;
}

export function heroTaglineForEntry(entry: WatchEntry | null, hasQuickLive: boolean): string {
  if (entry && isInProgress(entry)) {
    return continueLabel(entry);
  }
  if (hasQuickLive) return "Συνέχεια ζωντανά";
  return "Ξεκίνα από Live TV";
}

File: apps/android/app/src/main/java/com/apostolos/tv/ui/common/WatchEntryFormat.kt
package com.apostolos.tv.ui.common

import com.apostolos.tv.data.model.WatchEntry
import com.apostolos.tv.data.model.WatchType

fun formatWatchTime(ms: Long): String {
    if (ms <= 0L) return "0:00"
    val totalSeconds = ms / 1000
    val hours = totalSeconds / 3600
    val minutes = (totalSeconds % 3600) / 60
    val seconds = totalSeconds % 60
    return if (hours > 0) {
        String.format("%d:%02d:%02d", hours, minutes, seconds)
    } else {
        String.format("%d:%02d", minutes, seconds)
    }
}

fun WatchEntry.continueLabel(): String = when (type) {
    WatchType.MOVIE -> "Συνέχισε · ${formatWatchTime(positionMs)}"
    WatchType.SERIES_EPISODE -> {
        val base = subtitle.ifBlank { "Επεισόδιο" }
        "$base · ${formatWatchTime(positionMs)}"
    }
    WatchType.LIVE -> title
}

fun WatchEntry.remainingLabel(): String {
    if (durationMs <= 0L) return ""
    val remaining = (durationMs - positionMs).coerceAtLeast(0L)
    return "Απομένουν ${formatWatchTime(remaining)}"
}
